### ISSUE TITLE
[Workflow] Mark registry as internal and deprecate the service

### ISSUE THREAD

**Issue by fabpot** [symfony/symfony-docs#17160]
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#46000
| PR author(s) | @lyrixx
| Merged in    | 6.2

**Comment by lyrixx** [comment]
TL;DR:

1. The WorkflowRegistry should never be used (we must not document it)
2. People have to inject the workflow where they need it. Typing `WorkflowInterface` works fine if only one workflow exists. Otherwise use named serviced. [Example](https://github.com/lyrixx/SFLive-Paris2016-Workflow/blob/master/src/Controller/ArticleController.php#L59). This exists for ages and works fine. Otherwise, one can still use the raw service id : `workflow.[name]` or `state_machine.[name]`
3. If they need all workflows (docs for examples)  they can inject all services tagged with
    * `workflow` => all workflow + state_machine
    * `workflow.workflow` => all workflow
    * `workflow.state_machine` => all state_machine

**Comment by wouterj** [comment]
Hi @lyrixx. Can you make the suggested changes in this section https://symfony.com/doc/current/workflow.html#accessing-the-workflow-in-a-class  ? I think what is needed is remove the "Alternatively, ..." part and add a quick note about the services providing all workflows/state machines.
